app/src/main/java/com/lightricks/feedexercise/ui/feed/FeedViewModel.kt

```diff
@@ -33,8 +36,35 @@ open class FeedViewModel : ViewModel() {
         refresh()
     }
 
+    @SuppressLint("CheckResult")
     fun refresh() {
-        //todo: fix the implementation
+        NetworkAccess.feedApiService.getFeed()
+            .subscribeOn(Schedulers.io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({ feedResponse ->
+                handleResponse(feedResponse)
+            },{ error ->
+                handleNetworkError(error)
+            })
+    }
+
+    fun handleResponse(feedResponse: MetadataItemsList) {
+        convertFeedResponseToFeedItemList(feedResponse)
+    }
+
+    fun handleNetworkError(error: Throwable) {
+        Log.e("Test: ", error.message.toString())
```

Comment:
Here you should display and error to the user by assigning a value to the `networkErrorEvent` livedata.